src/main/java/stsjorbsmod/patches/HealthBarRenderSnapPatch.java

```diff
@@ -0,0 +1,64 @@
+package stsjorbsmod.patches;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.math.MathUtils;
+import com.evacipated.cardcrawl.mod.stslib.patches.powerInterfaces.HealthBarRenderPowerPatch;
+import com.evacipated.cardcrawl.modthespire.lib.*;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.helpers.ImageMaster;
+import com.megacrit.cardcrawl.powers.IntangiblePower;
+import javassist.CtBehavior;
+import stsjorbsmod.characters.Wanderer;
+import stsjorbsmod.memories.MemoryManager;
+import stsjorbsmod.powers.FragilePower;
+
+public class HealthBarRenderSnapPatch {
```

Comment:
I'm not sure I understand why patching the patch is necessary; can we not make `FragilePower` a `HealthBarRendererPower` instead, something like this?

```java
    @Override
    public int getHealthBarAmount() {
        if (this.amount != 1) { return 0; }
        int selfDamage = amount2 * SnapAction.PLAYER_DAMAGE_PER_CLARITY;
        if (owner.hasPower(IntangiblePlayerPower.POWER_ID)) {
            selfDamage = MathUtils.clamp(selfDamage, 0, 1);
        }
        return selfDamage;
    }

    @Override
    public Color getColor() {
        return new Color(
                (MathUtils.cosDeg((float) (System.currentTimeMillis() / 10L % 360L)) + 1.25F) / 2.3F,
                (MathUtils.cosDeg((float) ((System.currentTimeMillis() + 1000L) / 10L % 360L)) + 1.25F) / 2.3F,
                (MathUtils.cosDeg((float) ((System.currentTimeMillis() + 2000L) / 10L % 360L)) + 1.25F) / 2.3F,
                1.0F
        );
    }
```

Reply:
Yeah, you're right. I believe the reason I had patched the patch initially was because I wasn't using `FragilePower` initially and then just didn't rethink the approach.